### ISSUE TITLE
Question for training details 

### ISSUE THREAD

**Issue by pluiez** [kyutai-labs/moshi#111]
### Due diligence

- [X] I have done my due diligence in trying to find the answer myself.

### Topic

The paper

### Question

Hi, dear authors

1. Moshi pretraining and post-training
The two stages both involve a portion of text-only for language modeling training to preserve lanauge capability. I'm wondering which component produces the logit for next text token, is it the orginal LM head of Helium or the first prediciton of depth transformer?

2. Moshi pretraining
In the paper you mentioned:
> We mask the corresponding text tokens with a probability of 30%.
Since we are doing generative modeling instead of BERT-style masked prediction, I don't quite understand what this means and why this is necessary 

3. Multi-stream training
Starting from Moshi Post-training stage, the model continues to train on multi-stream data. In Figure 4, It's obvious that at inference time, the model always predicts "text token -> moshi stream" and ignores the user stream. But at training time, do you optimize on "text token -> moshi stream -> user stream"? i.e., do you calculate the loss of predicting user stream tokens?

4. Initialization for Moshi pretraining
Moshi is initialized from Helium, an un-aligned language model incapable of basic conversation and instruction following. Have you tried a Helium-instruct model to initialize single audio stream training? Maybe this can reduce the loss of language capability with following audio finetunings.

Thanks!

**Comment by TianwenWei** [comment]
For the first question, it can be seen from the code that it is the temporal transformer that produces the logit for the next text token.
